src/components/Sidebar/CommentList/CommentItem.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext } from 'react';
+import { Comment } from '../../../types/Comment';
+import { CommentListContext } from '../../../utils/Store';
+import { deleteComment } from '../../../utils/fetchFunctions';
+
+interface CommentItemProps {
+  comment: Comment;
+}
+
+export const CommentItem: React.FC<CommentItemProps> = ({ comment }) => {
+  const { comments, setComments } = useContext(CommentListContext);
+
+  async function handleDelete() {
+    const newList = comments.filter(item => item.id !== comment.id);
+    const copy = [...comments];
+
+    setComments(newList);
+
+    try {
+      await deleteComment(comment.id);
+    } catch (error) {
+      alert('Failed to delete comment, please try again!');
+
+      setComments(copy);
```

Comment:
Consider using a more descriptive error handling mechanism instead of `alert`, which can be disruptive to the user experience. You might want to display an error message within the UI instead.